x-pack/plugins/observability_ai_assistant/public/functions/lens.tsx

```diff
@@ -263,6 +271,8 @@ export function registerLensFunction({
         },
       });
 
+      if (!timeField) return;
```

Comment:
should we not default to `@timestamp` in this case? Have you seen the LLM respond with an empty string?

Reply:
for some reason lens function gets executed multiple times sometimes, where first time (or not the last time) is missing properties. seems to be a bug somewhere else in our rendering, but i wasn't able to find it, so i added a check here to make sure we only render the chart once timefield is set.

timefield should actually never be empty as it has default value set.

Reply:
> for some reason lens function gets executed multiple times sometimes, where first time (or not the last time) is missing properties. seems to be a bug somewhere else in our rendering, but i wasn't able to find it, so i added a check here to make sure we only render the chart once timefield is set.

Not sure if I follow. Why is it a problem if it renders multiple times? (I assume it renders multiple times, but to the same element)

Reply:
not on its own, but lens seems to be doing multiple requests then, which is a performance hit if nothing else. i think its safe to not render until we get all the required properties